sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/MeterSelector.java

```diff
@@ -0,0 +1,98 @@
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Provides means for selecting one or more Meters. Used for selecting instruments when constructing
+ * views.
+ */
+@AutoValue
+@Immutable
+public abstract class MeterSelector {
+
+  /**
+   * Returns a new {@link Builder} for {@link InstrumentSelector}.
+   *
+   * @return a new {@link Builder} for {@link InstrumentSelector}.
+   */
+  public static Builder builder() {
+    return new AutoValue_MeterSelector.Builder()
+        .setNameFilter(StringPredicates.ALL)
+        .setVersionFilter(StringPredicates.ALL)
+        .setSchemaUrlFilter(StringPredicates.ALL);
+  }
+
+  /**
+   * Returns the {@link Pattern} generated by the provided {@code regex} in the {@link Builder}, or
+   * {@code Pattern.compile(".*")} if none was specified.
+   */
+  public abstract Predicate<String> getNameFilter();
+
+  /**
+   * Returns the {@link Pattern} generated by the provided {@code regex} in the {@link Builder}, or
+   * {@code Pattern.compile(".*")} if none was specified.
+   */
+  public abstract Predicate<String> getVersionFilter();
+
+  /**
+   * Returns the {@link Pattern} generated by the provided {@code regex} in the {@link Builder}, or
+   * {@code Pattern.compile(".*")} if none was specified.
+   */
+  public abstract Predicate<String> getSchemaUrlFilter();
+
+  /** Builder for {@link InstrumentSelector} instances. */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /** Sets the {@link Predicate} for matching name. */
+    abstract Builder setNameFilter(Predicate<String> filter);
+
+    /** Sets the {@link Pattern} for matching name. */
+    public final Builder setNamePattern(Pattern pattern) {
+      return setNameFilter(StringPredicates.regex(pattern));
+    }
+
+    /** Sets a specifier for selecting Instruments by name. */
+    public final Builder setName(String name) {
+      return setNameFilter(StringPredicates.exact(name));
+    }
+
+    /** Sets the {@link Predicate} for matching versions. */
+    abstract Builder setVersionFilter(Predicate<String> filter);
+
+    /** Sets the {@link Pattern} for matching versions. */
+    public final Builder setVersionPattern(Pattern pattern) {
+      return setVersionFilter(StringPredicates.regex(pattern));
+    }
+
+    /** Sets a specifier for selecting Meters by version. */
+    public final Builder setVersion(String version) {
+      return setVersionFilter(StringPredicates.exact(version));
+    }
+
+    /** Sets the {@link Predicate} for matching schema urls. */
+    abstract Builder setSchemaUrlFilter(Predicate<String> filter);
+
+    /** Sets the {@link Pattern} for matching schema urls. */
+    public final Builder setSchemaUrlPattern(Pattern pattern) {
+      return setSchemaUrlFilter(StringPredicates.regex(pattern));
+    }
+
+    /** Sets a specifier for selecting Instruments by name. */
+    public final Builder setSchemaUrl(String url) {
+      return setSchemaUrlFilter(StringPredicates.exact(url));
+    }
```

Comment:
do we need both of these?

Reply:
Switched to this from one of @anuraaga's comments on performance.   You can do exact matching w/ regex, but it's clunky.  `Pattern.compile(Pattern.quote(input))`.  I can switch back, it's not a huge deal either way

Reply:
ah, yeah. hmm. I don't think view lookup should be a hot-path thing, though, so I'm not so worried about performance, personally. Unless I'm mistaken that it might be on the users hot-path somewhere?

Reply:
It *could* be on the hot-path.  Instruments can be pulled off meters at any point in the lifecycle of the meter.  Depending on instrumentation, this could be as frequent as once-per-request.  While we recommend storing these somewhere more static / longer-lived, I think the predicate-based code (once I saw it) will be easier to maintain/extend.

If it's all the same, I'd like to use the predicate solution here (and elsewhere). Happy to remove some public methods if that simplifies things.